projects/packages/search/src/class-cli.php

```diff
@@ -19,15 +20,47 @@ class CLI extends WP_CLI_Command {
 	 *
 	 * ## EXAMPLES
 	 *
-	 * wp jetpack-search auto_config
+	 * wp jetpack-search auto_config user_login
+	 *
+	 * wp jetpack-search auto_config user_id
+	 *
+	 * @param array $args - Args passsed in.
 	 */
-	public function auto_config() {
+	public function auto_config( $args ) {
 		try {
+			if ( empty( $args ) ) {
+				WP_CLI::error( 'A user login or ID is required.' );
+			}
+
+			// Some functions may require admin capabilities to run.
+			$ret = $this->set_user( $args[0] );
+			if ( is_wp_error( $ret ) ) {
+				WP_CLI::error( $ret->get_error_message() );
+			}
+
+			WP_CLI::line( 'Running as user ' . $ret->user_login . '…' );
 			$blog_id = Helper::get_wpcom_site_id();
 			Instant_Search::instance( $blog_id )->auto_config_search();
-			WP_CLI::line( 'Jetpack Search: auto config success!' );
+			WP_CLI::success( 'Auto config success!' );
```

Comment:
Something for the future: It'd be nice if the CLI were more verbose about what it's doing in the future. It'd be nice if it printed something along these lines, for example:

> Auto-configuring Jetpack Search...
> - Search module is **already enabled**.
> - Instant Search **enabled**.
> - The theme sidebar **already contains a Jetpack Search widget**, ignoring.
> - The overlay sidebar has been **initialized with a Jetpack Search widget**.